ompi/mca/coll/accelerator/coll_accelerator.h

```diff
@@ -45,6 +46,11 @@ mca_coll_accelerator_allreduce(const void *sbuf, void *rbuf, size_t count,
                         struct ompi_communicator_t *comm,
                         mca_coll_base_module_t *module);
 
+int mca_coll_accelerator_reduce_local(const void *sbuf, void *rbuf, size_t count,
```

Comment:
FWICT `mca_coll_accelerator_reduce_local` only differs from `mca_coll_accelerator_reduce` in the absence of `root`. I wonder if we really need a special API for it.

How about introducing a const value `#define MCA_COLL_ACCELERATOR_LOCAL_ROOT -1` instead of creating a new interface?

Another alternative is to put this in opal/mca/accelerator - the local reduction is arguably an accelerator capability rather than communicator behavior.

@devreal should be a better person to comment on this.

Reply:
I think the issue is that reduce_local is a separate function pointer on the coll structure, so we need to have a function that we can register with the component. It has to be in the component, so we cannot use a function from the accelerator framework API, that would be an abstraction violation.

Reply:
@edgargabriel is correct. We need a separate function to point to that doesn't include (comm, root) arguments. This is why we can't reuse existing reduce function.